File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/command/source/CommandSource.java
package pize.tests.voxelgame.main.command.source;

import pize.math.vecmath.vector.Vec3f;
import pize.tests.voxelgame.main.text.Component;
import pize.tests.voxelgame.server.player.ServerPlayer;

public abstract class CommandSource{
    
    public abstract Vec3f getPosition();
    
    public abstract void sendMessage(Component message);
    
    
    public ServerPlayer asPlayer(){
        return ((CommandSourcePlayer) this).getPlayer();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/command/source/CommandSource.kt
package pize.tests.voxelgame.main.command.source

import pize.math.vecmath.vector.Vec3f
import pize.tests.voxelgame.main.text.Component
import pize.tests.voxelgame.server.player.ServerPlayer

abstract class CommandSource {
    abstract val position: Vec3f
    abstract fun sendMessage(message: Component?)
    fun asPlayer(): ServerPlayer? {
        return (this as CommandSourcePlayer).player
    }
}
